### PR TITLE
Test api errors thrown up

### PR DESCRIPTION
**Pull request by willmarks**
## Description

Add explicit tests to confirm that errors are being thrown out of readRecords.

Probably not worth bumping the version for this.

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/servicenow-source/test/index.test.ts [hunk 1]
```diff
@@ -118,6 +118,22 @@ describe('index', () => {
...
+    const streams = source.streams({});
+    const stream = streams[0];
+    try {
+      const iter = stream.readRecords(SyncMode.FULL_REFRESH);
+      await iter.next();
+    } catch (err: any) {
+      expect(err.message).toBe(expectedMessage);
+      return;
+    }
+    expect(0).toBe(1);
```
Comment (tovbinm) [inline sources/servicenow-source/test/index.test.ts:134]: remove `expect(0).toBe(1);`

File: sources/servicenow-source/test/index.test.ts [hunk 2]
```diff
@@ -153,4 +169,20 @@ describe('index', () => {
...
+    const streams = source.streams({});
+    const stream = streams[1];
+    try {
+      const iter = stream.readRecords(SyncMode.FULL_REFRESH);
+      await iter.next();
+    } catch (err: any) {
+      expect(err.message).toBe(expectedMessage);
+      return;
+    }
+    expect(0).toBe(1);
```
Comment (tovbinm) [inline sources/servicenow-source/test/index.test.ts:186]: same

File: sources/servicenow-source/test/index.test.ts (hunk 1 above)
Comment (willmarks) [reply sources/servicenow-source/test/index.test.ts:134]: If an error is not thrown/caught then the test passes when it should fail.

Comment (tovbinm) [reply sources/servicenow-source/test/index.test.ts]: I see, then do 
```
expect(() => {
      const iter = stream.readRecords(SyncMode.FULL_REFRESH);
      await iter.next();
      fail('Error should have been thrown.');
}).toThrow(....)
```

Comment (tovbinm) [reply sources/servicenow-source/test/index.test.ts]: fail also looks ok.

**Review by tovbinm — APPROVED** [review]

**Pull request merged**